datafusion/core/tests/sql/select.rs

```diff
@@ -1210,3 +1210,30 @@ async fn unprojected_filter() {
     ];
     assert_batches_sorted_eq!(expected, &results);
 }
+
+#[tokio::test]
+async fn case_sensitive_in_default_dialect() {
+    let int32_array = Int32Array::from(vec![1, 2, 3, 4, 5]);
+    let schema = Schema::new(vec![Field::new("INT32", DataType::Int32, false)]);
+    let batch =
+        RecordBatch::try_new(Arc::new(schema), vec![Arc::new(int32_array)]).unwrap();
+
+    let ctx = SessionContext::new();
+    let table = MemTable::try_new(batch.schema(), vec![vec![batch]]).unwrap();
+    ctx.register_table("t", Arc::new(table)).unwrap();
+    let sql = "select \"INT32\" from t";
+    let actual = execute_to_batches(&ctx, sql).await;
+
+    let expected = vec![
+        "+-------+",
+        "| INT32 |",
+        "+-------+",
+        "| 1     |",
+        "| 2     |",
+        "| 3     |",
+        "| 4     |",
+        "| 5     |",
+        "+-------+",
+    ];
+    assert_batches_eq!(expected, &actual);
+}
```

Comment:
Looks good, thank you @Rachelint 

it would be great if we could add another test case for "SELECT "int32" FROM t" and make sure it output the error like "cannot find Column int32"

Reply:
> Looks good, thank you @Rachelint
> 
> it would be great if we could add another test case for "SELECT "int32" FROM t" and make sure it output the error like "cannot find Column int32"

Thanks, I have added it.

Reply:
LGTM -- thanks @Rachelint